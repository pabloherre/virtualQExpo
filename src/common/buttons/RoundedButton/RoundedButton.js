import React, {Component} from 'react';
import {TouchableOpacity, Text} from 'react-native';

class RoundedButton extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <TouchableOpacity style={styles.button} {...this.props}>
        <Text style={styles.buttonText}>{this.props.label}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = {
  button: {
    backgroundColor: '#F77027',
    borderRadius: 30,
    paddingTop: 10,
    paddingBottom: 10,
    alignItems: 'center',
    justifyContent: 'center',
    height: 60,
  },
  buttonText: {
    fontSize: 20,
    color: 'white',
  },
};

export default RoundedButton;
