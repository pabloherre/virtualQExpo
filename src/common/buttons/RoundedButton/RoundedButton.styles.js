import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  button: {
    backgroundColor: '#F77027',
    borderRadius: 30,
    paddingVertical: 10,
    paddingHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    height: 60
  },

  buttonText: {
    paddingHorizontal: 10,
    fontSize: 20,
    color: 'white'
  }
});
