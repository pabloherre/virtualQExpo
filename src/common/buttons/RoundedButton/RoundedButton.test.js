import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RoundedButton from './RoundedButton';

it('renders correctly', () => {
  const tree = renderer.create(<RoundedButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('can press button', () => {
  const onPress = jest.fn();
  const button = shallow(<RoundedButton onPress={onPress} />);
  button.dive().simulate('press');
  expect(onPress).toHaveBeenCalled();
});

it('has label', () => {
  const button = shallow(<RoundedButton label={'algo'} />);
  const text = button.childAt(0);
  expect(text.text()).toEqual();
});
