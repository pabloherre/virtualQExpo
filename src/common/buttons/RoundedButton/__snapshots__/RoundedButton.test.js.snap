// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundedButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f77027",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-around",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "paddingHorizontal": 10,
      }
    }
  >
    test
  </Text>
</View>
`;
